### PR TITLE
Added support for Links filtering

### PR DESCRIPTION
**Pull request by kiridarivaki**
References #5741

### CODE REVIEW

**Comment by charlesBochet** [comment]
@kiridarivaki  Thanks for your PR :) Honnestly, I think we should restrict the scope to supporting Links filtering. Supporting sort in this PR (for Thursday) looks very ambitious. Let's treat that in a separated PR and keep this one for Links filtering

1) Please run `npx nx run twenty-front:lint:ci`, you should see some errors. (I believe they are tied to the changes you have made for sorting)
2) The work on Links looks good but I cannot test it as the front does not build at the moment
3) we'll have few lines to add in the backend to remove secondaryLinks from the FilterInput
image

**Comment by kiridarivaki** [comment]
thanks! i'll focus on filtering from now on

**Comment by kiridarivaki** [comment]
@charlesBochet linting passes when i run ```npx nx run twenty-front:lint:ci``` so i can't see any errors (also i haven't pushed any changes for sorting). I just added the be code :)

**Comment by kiridarivaki** [comment]
@charlesBochet linting passes when i run npx nx run twenty-front:lint:ci so i can't see any errors (also i haven't pushed any changes for sorting). I just added the be code :)

**Comment by charlesBochet** [comment]
@kiridarivaki Sorry, I've mistaken your PR for another. taking another look

File: packages/twenty-server/src/engine/api/graphql/workspace-schema-builder/factories/composite-input-type-definition.factory.ts [hunk 1]
```diff
@@ -62,6 +62,14 @@ export class CompositeInputTypeDefinitionFactory {
         continue;
       }
 
+      // Skip secondaryLinks property for LINKS type in Filter input type
+      if (
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/api/graphql/workspace-schema-builder/factories/composite-input-type-definition.factory.ts:66]: can we actually try to change packages/twenty-server/src/engine/metadata-modules/field-metadata/composite-types/links.composite-type.ts to have hidden: 'input' on secondaryLinks. This should leverage the continue check line 61 of composite-input-type-definition.factory and avoid adding this extra edge case

**Review by charlesBochet — COMMENTED** [review]
Thank you! Looks great actually, left another comment

**Review by charlesBochet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**